pkg/bootkube/parse.go

```diff
@@ -0,0 +1,67 @@
+package bootkube
+
+import (
+	"io/ioutil"
+	"path/filepath"
+	"strings"
+
+	"github.com/ghodss/yaml"
+	"github.com/kubernetes-incubator/bootkube/pkg/asset"
+	"k8s.io/kubernetes/pkg/apis/extensions/v1beta1"
+)
+
+const (
+	defaultServiceCIDR = "10.3.0.0/24"
+	defaultPodCIDR     = "10.2.0.0/24"
+)
+
+// detectServiceCIDR deserializes the '--service-cluster-ip-range' from the
+// kube-apiserver manifest
+func detectServiceCIDR(config Config) string {
+	b, err := ioutil.ReadFile(filepath.Join(config.AssetDir, asset.AssetPathAPIServer))
+	if err != nil {
+		UserOutput("Can't detect --service-cluster-ip-range in %s, using %s: %v\n", asset.AssetPathAPIServer, defaultServiceCIDR, err)
+		return defaultServiceCIDR
+	}
+	var apiServer v1beta1.DaemonSet
+	yaml.Unmarshal(b, &apiServer)
+
+	containers := apiServer.Spec.Template.Spec.Containers
+	if len(containers) > 0 {
+		return findFlag("--service-cluster-ip-range", containers[0].Command)
+	}
+	UserOutput("Can't detect --service-cluster-ip-range, using %s\n", defaultServiceCIDR)
+	return defaultServiceCIDR
+}
+
+// detectPodCIDR deserializes the '--cluster-cidr' from the
+// kube-controller-manager manifest.
+func detectPodCIDR(config Config) string {
+	b, err := ioutil.ReadFile(filepath.Join(config.AssetDir, asset.AssetPathControllerManager))
+	if err != nil {
+		UserOutput("Can't detect --cluster-cidr in %s, using %s: %v\n", asset.AssetPathControllerManager, defaultPodCIDR, err)
+		return defaultPodCIDR
+	}
+	var manager v1beta1.Deployment
+	yaml.Unmarshal(b, &manager)
+
+	containers := manager.Spec.Template.Spec.Containers
+	if len(containers) > 0 {
+		return findFlag("--cluster-cidr", manager.Spec.Template.Spec.Containers[0].Command)
+	}
+	UserOutput("Can't detect --cluster-cidr, using %s\n", defaultPodCIDR)
+	return defaultPodCIDR
+}
+
+func findFlag(flagName string, args []string) string {
```

Comment:
Hm, flagsets (both stdlib and spf13) expect all flags to be registered, which means we need to do that stripping and also get the appropriate flags for an apiserver and controller manager (similar to https://github.com/kubernetes-incubator/bootkube/blob/master/pkg/bootkube/bootkube.go#L53). Doesn't seem like a great general flag parser unless I'm missing something.